pkg/utils/utils.go

```diff
@@ -185,17 +190,29 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
 }
 
 // SkipConfigVf Use systemd service to configure switchdev mode or BF-2 NICs in OpenShift
-func SkipConfigVf(ifSpec sriovnetworkv1.Interface, ifStatus sriovnetworkv1.InterfaceExt) bool {
+func SkipConfigVf(ifSpec sriovnetworkv1.Interface, ifStatus sriovnetworkv1.InterfaceExt) (bool, error) {
 	if ifSpec.EswitchMode == sriovnetworkv1.ESwithModeSwitchDev {
 		glog.V(2).Infof("SkipConfigVf(): skip config VF for switchdev device")
-		return true
+		return true, nil
 	}
+
 	// Nvidia_mlx5_MT42822_BlueField-2_integrated_ConnectX-6_Dx in OpenShift
 	if ClusterType == ClusterTypeOpenshift && ifStatus.Vendor == VendorMellanox && ifStatus.DeviceID == DeviceBF2 {
-		glog.V(2).Infof("SkipConfigVf(): skip config VF for BF2 device")
-		return true
+		// TODO: remove this when switch to the systemd configuration support.
+		mode, err := mellanoxBlueFieldMode(ifStatus.PciAddress)
+		if err != nil {
+			return false, fmt.Errorf("failed to read Mellanox Bluefield card mode for %s,%v", ifStatus.PciAddress, err)
+		}
+
+		if mode == bluefield2Conntexd {
+			return false, nil
+		}
+
+		glog.V(2).Infof("SkipConfigVf(): skip config VF for BF2 device on DPU mode")
```

Comment:
Neet to confirm that such use-case is not needed

Reply:
what is the use case you refer ?